components/movementsensor/gpsrtkpmtk/gpsrtkpmtk.go

```diff
@@ -682,3 +698,80 @@ func (g *rtkI2C) Close(ctx context.Context) error {
 
 	return nil
 }
+
+// sendGGAMessage sends GGA messages to the serial port. This is only used to get NTRIP stream
+// from a virtual reference point.
+func (g *rtkI2C) sendGGAMessage() {
+	// Open an I2C handle to read NMEA messages
+	handle, err := g.bus.OpenHandle(g.addr)
+	if err != nil {
+		g.logger.Errorf("can't open gps i2c handle: %s", err)
+		g.err.Set(err)
+		return
+	}
+	defer func() {
+		if closeErr := handle.Close(); closeErr != nil {
```

Comment:
:+1: Thanks for deferring closing the handle immediately! This avoids all kinds of bugs that might otherwise occur.